### PR TITLE
add `getTables` method

### PR DESCRIPTION
**Pull request by xuwei-k**
- fix https://github.com/scalikejdbc/scalikejdbc/issues/2258

### CODE REVIEW

**Review by seratch — APPROVED** [review]
Thanks, this looks great to me

File: scalikejdbc-core/src/main/scala/scalikejdbc/DBConnection.scala [hunk 1]
```diff
@@ -530,25 +542,27 @@ trait DBConnection extends LogSupport with LoanPattern with AutoCloseable {
...
             rs.string("TABLE_NAME"),
             rs.string("REMARKS")
           )
         }
         .to(LazyList)
 
-    tableList.headOption.map { case (schema, table, remarks) =>
+    tableList.map { case (catalog, schema, table, remarks) =>
       val pkNames: List[String] = new ResultSetIterator(
         meta.getPrimaryKeys(null, schema, table)
```
Comment (xuwei-k) [inline scalikejdbc-core/src/main/scala/scalikejdbc/DBConnection.scala:560]: - https://github.com/scalikejdbc/scalikejdbc/pull/2261

**Comment by seratch** [comment]
@xuwei-k Would it be feasible to apply the same change to 4.3.x series? If yes, I am happy to cut a new patch release.

**Comment by xuwei-k** [comment]
- https://github.com/scalikejdbc/scalikejdbc/pull/2262

**Pull request merged**